sympy/core/expr.py

```diff
@@ -3406,12 +3419,23 @@ def round(self, p=0):
             rv = -(Integer(-i10)//10)
         else:
             rv = Integer(i10)//10
-        q = 1
-        if p > 0:
-            q = mag
-        elif p < 0:
-            rv /= mag
-        rv = Rational(rv, q)
+        rv = Rational(rv, mag)
+        if how == 'even':
+            # rv >= self*mag
+            i = rv*mag
+            if 2*(i - self*mag) == 1 and i%2:
+                # round toward even
+                rv = Rational(i - 1, mag)
+        elif how != 'up':
```

Comment:
"up" is ambiguous. Does this mean towards +infinity or away from 0?

Reply:
I vote for not supporting custom rounding. That way the round function and method are synonymous in behavior.

Reply:
But to answer the question, it rounds away from 0.

Reply:
That's fine too. There are many rounding methods which could be added, but they don't need to be done in this PR.